// intasend-node/dist/payouts.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var RequestClient = require('./requests');
var Payouts = /** @class */ (function (_super) {
    __extends(Payouts, _super);
    function Payouts() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    Payouts.prototype.initiate = function (payload) {
        return this.send(payload, '/api/v1/send-money/initiate/', 'POST');
    };
    Payouts.prototype.mpesa = function (payload) {
        payload['provider'] = 'MPESA-B2C';
        return this.initiate(payload);
    };
    Payouts.prototype.mpesaB2B = function (payload) {
        payload['provider'] = 'MPESA-B2B';
        return this.initiate(payload);
    };
    Payouts.prototype.bank = function (payload) {
        payload['provider'] = 'PESALINK';
        return this.initiate(payload);
    };
    Payouts.prototype.intasend = function (payload) {
        payload['provider'] = 'INTASEND';
        return this.initiate(payload);
    };
    Payouts.prototype.airtime = function (payload) {
        payload['provider'] = 'AIRTIME';
        return this.initiate(payload);
    };
    Payouts.prototype.approve = function (payload) {
        return this.send(payload, '/api/v1/send-money/approve/', 'POST');
    };
    Payouts.prototype.status = function (payload) {
        return this.send(payload, '/api/v1/send-money/status/', 'POST');
    };
    return Payouts;
}(RequestClient));
module.exports = Payouts;
