// intasend-node/dist/wallets.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var RequestClient = require('./requests');
var Wallet = /** @class */ (function (_super) {
    __extends(Wallet, _super);
    function Wallet() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    Wallet.prototype.list = function () {
        return this.send({}, '/api/v1/wallets/', 'GET');
    };
    Wallet.prototype.create = function (payload) {
        return this.send(payload, '/api/v1/wallets/', 'POST');
    };
    Wallet.prototype.intraTransfer = function (sourceID, destinationID, amount, narrative) {
        var payload = {
            wallet_id: destinationID,
            amount: amount,
            narrative: narrative,
        };
        return this.send(payload, "/api/v1/wallets/".concat(sourceID, "/intra_transfer/"), 'POST');
    };
    Wallet.prototype.get = function (walletID) {
        return this.send({}, "/api/v1/wallets/".concat(walletID, "/"), 'GET');
    };
    Wallet.prototype.transactions = function (walletID) {
        return this.send({}, "/api/v1/wallets/".concat(walletID, "/transactions/"), 'GET');
    };
    Wallet.prototype.fundMPesa = function (payload) {
        payload['method'] = 'M-PESA';
        payload['currency'] = 'KES';
        return this.send(payload, '/api/v1/payment/mpesa-stk-push/', 'POST');
    };
    Wallet.prototype.fundCheckout = function (payload) {
        this.secret_key = '';
        return this.send(payload, '/api/v1/checkout/', 'POST');
    };
    return Wallet;
}(RequestClient));
module.exports = Wallet;
